app/helpers/repositories_helper.rb

```diff
@@ -271,6 +271,50 @@ def show_settings_save_button?(_repository)
       !@repository.managed?
   end
 
+  def breadcrumbs(project:, path:, rev:)
+    [
+      project_breadcrumb(project),
+      repository_breadcrumb(project, rev),
+      *path_breadcrumbs(project, path, rev)
+    ]
+  end
+
+  def project_breadcrumb(project)
+    {
+      href: project_overview_path(project.id),
+      text: project.name
+    }
+  end
+
+  def repository_breadcrumb(project, rev)
+    {
+      href: url_for(action: "show", project_id: project, repo_path: nil, rev: rev),
```

Comment:
I stumbled over `project_id: project`. Shouldn't it rather be:

```suggestion
      href: url_for(action: "show", project_id: project.id, repo_path: nil, rev: rev),
```